tutorials/3d/portals/advanced.rst

```diff
@@ -0,0 +1,96 @@
+
+.. image:: img/roomgroups.png
+
+For instance, when outside you may wish to use a :ref:`DirectionalLight<class_DirectionalLight>` to represent the sun. When the outside RoomGroup receives an ``enter gameplay`` callback, you can turn the light on, and you can turn it off when the RoomGroup exits gameplay. With the light off, performance will increase as there is no need to render it indoors.
+
+This is an example of a simple RoomGroup script to turn on and off a DirectionalLight (note that you can also use signals for callbacks, the choice is up to you) :
+
+.. image:: img/roomgroup_notification.png
+
+.. tip:: You can apply the same technique for switching on and off rain effects, skyboxes and much more.
+
+Internal Rooms
+~~~~~~~~~~~~~~
+There is one more trick that RoomGroups have up their sleeve. A very common desire is to have a game level with a mixed outdoor and indoor environment. We have already mentioned that rooms can be used to represent both rooms in a building, and areas of landscape, such as a canyon.
+
+What happens if you wish to have a house in a terrain 'room'?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+With the functionality described so far you *can* do it - you would need to place portals around the exterior of the house though, forming needless rooms above the house. This has been done in many games. But what if there was a simpler way?
+
+It turns out there is a simpler way of handling this scenario. Godot supports *rooms WITHIN rooms* (we will call them **'internal rooms'**). That is, you can place a house within a terrain room, or even a building, or set of buildings, and even have exit portals in different terrain rooms!
+
+This is actually very simple to do. You don't need to place a room within another room in the scene tree (in fact you will get a warning if you try this). Instead, just create them as regular rooms. But the internal rooms should be grouped together with a RoomGroup as parent. If you look in the inspector for the RoomGroup there is a ``roomgroup_priority`` which defaults to ``0``.
+
+If you want a room or set of rooms to be internal, just set the priority to a higher value than the outer (enclosing) room, using the RoomGroup. That's all there is to it.
+
+The system uses the priority setting to give priority to the internal room when deciding which room a camera or object is within - higher priority always wins. Everything else works in a mostly similar way.
+
+The only differences:
+
+- Portals between internal rooms and outer rooms should always *be placed in the inner (internal) room*
+- Portals of internal rooms are not considered as part of the bound of outer rooms
+- ``STATIC`` and ``DYNAMIC`` Objects from outer rooms will not sprawl into internal rooms. If you want objects to cross these portals, place them in the internal room. This is to prevent large objects like terrain sections sprawling into entire buildings, and rendering when not necessary.
```

Comment:
```suggestion
The only differences:

- Portals between internal rooms and outer rooms should always *be placed in the inner (internal) room*.
- Portals of internal rooms are not considered as part of the bound of outer rooms.
- ``STATIC`` and ``DYNAMIC`` objects from outer rooms will not sprawl into internal rooms. If you want objects to cross these portals, place them in the internal room. This is to prevent large objects like terrain sections sprawling into entire buildings, and rendering when not necessary.
```